docs/modules/develop/pages/guide_development_with_localhost_ssl.adoc

```diff
@@ -0,0 +1,63 @@
+= Testing SSL in localhost
+
+In some cases is useful to access the `localhost` environment under SSL (ie: `https://localhost:3000/`).
+
+A typical use case is when you want to test login integration with 3rd party systems using OAuth (which in the latests versions only the HTTPS protocol is allowed).
+
+Using Puma, the default server for Rails, you can easily do the trick by creating a self-signed certificate and using it when starting your development environment.
+
+== Create a self-signed certificate for localhost
+
+First you need a SSL certificate for the `localhost` domain (you only need to execute this command once):
+
+[source,bash]
+----
+openssl req -x509 -out localhost.crt -keyout localhost.key \
+  -newkey rsa:2048 -nodes -sha256 \
+  -subj '/CN=localhost' -extensions EXT -config <( \
+   printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
+----
+
+This will generate two files: `localhost.crt` and `localhost.key`, you can store them anywhere you want to use them later. In this example we'll suppose we have them in `/home/alice/certs/`:
+
+[source,bash]
+----
+mkdir /home/alice/certs/
+mv localhost.crt /home/alice/certs/
+mv localhost.key /home/alice/certs/
+----
+
+== Starting development in SSL mode
+
+Once you have your certificate is generated, just use the next command in order to start puma instead of the typical `bin/rails s`:
+
+[source,bash]
+----
+bin/rails s -b 'ssl://127.0.0.1:3000?key=/home/alice/certs/localhost.key&cert=/home/alice/certs/localhost.crt'
```

Comment:
Accepting the previous comment implies accepting this:

```suggestion
bin/rails s -b 'ssl://127.0.0.1:3000?key=~/alice/certs/localhost.key&cert=~/alice/certs/localhost.crt'
```

Reply:
Unfortunately, the `symbol ~` is not resolved by puma, let me, try an alternative